pkg/machinery/resources/network/dns_server.go

```diff
@@ -0,0 +1,65 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package network
+
+import (
+	"net/netip"
+
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/resource/meta"
+	"github.com/cosi-project/runtime/pkg/resource/protobuf"
+	"github.com/cosi-project/runtime/pkg/resource/typed"
+
+	"github.com/siderolabs/talos/pkg/machinery/proto"
+)
+
+// DNSServerType is type of DNSServer resource.
+const DNSServerType = resource.Type("DNSServers.net.talos.dev")
+
+// DNSServer resource holds DNS resolver info.
+type DNSServer = typed.Resource[DNSServerSpec, DNSServerExtension]
```

Comment:
I think this should be `ResolverStatus`, but we have this already occupied.

`DNSServer` is not really an accurate name, as we don't run an authoritative DNS server, but rather a resolver which forwads the requests. Probably something more about `resolved` ? And same for the controller name possibly. Need some clever name here.

Reply:
It's `DNSResolveCache` now. I think it sounds pretty nice.